automated OLM test case OCP-22070

jianzhangbjz opened the pull request:
Automated [test case OCP-22070](https://polarion.engineering.redhat.com/polarion/#/project/OSE/workitem?id=OCP-22070). @ecordell @njhale Could you help review it? Thanks very much!  cc: @bandrade @scolange @cuipinghuo @dongboyan77 @zihantang-rh @emmajiafan @chengzhang1016 

The logs as below:
```console
mac:origin jianzhang$ openshift-tests run all --dry-run | grep "\[Feature:Platform\] OLM should"  | openshift-tests run --junit-dir=./ -f -
started: (0/1/7) "[Feature:Platform] OLM should list clusterserviceversions [Suite:openshift/conformance/parallel]"

started: (0/2/7) "[Feature:Platform] OLM should list subscriptions [Suite:openshift/conformance/parallel]"

started: (0/3/7) "[Feature:Platform] OLM should list packagemanifests [Suite:openshift/conformance/parallel]"

started: (0/4/7) "[Feature:Platform] OLM should list catalogsources [Suite:openshift/conformance/parallel]"

started: (0/5/7) "[Feature:Platform] OLM should [ocp-22070]support grpc sourcetype [Suite:openshift/conformance/parallel]"

started: (0/6/7) "[Feature:Platform] OLM should list installplans [Suite:openshift/conformance/parallel]"

started: (0/7/7) "[Feature:Platform] OLM should list operatorgroups [Suite:openshift/conformance/parallel]"

I0617 20:02:51.206867   25871 trace.go:81] Trace[80895438]: "Reflector github.com/openshift/origin/pkg/monitor/pod.go:254 ListAndWatch" (started: 2019-06-17 20:02:33.549146 +0800 CST m=+9.794528807) (total time: 17.657155844s):
Trace[80895438]: [17.656849635s] [17.656849635s] Objects listed
passed: (38.3s) 2019-06-17T12:03:11 "[Feature:Platform] OLM should list subscriptions [Suite:openshift/conformance/parallel]"

passed: (40.7s) 2019-06-17T12:03:14 "[Feature:Platform] OLM should list installplans [Suite:openshift/conformance/parallel]"

passed: (42.5s) 2019-06-17T12:03:16 "[Feature:Platform] OLM should list catalogsources [Suite:openshift/conformance/parallel]"

passed: (46.2s) 2019-06-17T12:03:19 "[Feature:Platform] OLM should list clusterserviceversions [Suite:openshift/conformance/parallel]"

passed: (47.6s) 2019-06-17T12:03:21 "[Feature:Platform] OLM should list operatorgroups [Suite:openshift/conformance/parallel]"

passed: (51.4s) 2019-06-17T12:03:24 "[Feature:Platform] OLM should list packagemanifests [Suite:openshift/conformance/parallel]"

passed: (2m44s) 2019-06-17T12:05:17 "[Feature:Platform] OLM should [ocp-22070]support grpc sourcetype [Suite:openshift/conformance/parallel]"


Timeline:

Jun 17 12:02:40.351 E kube-apiserver Kube API started failing: Get https://api.qe-jian-0617.qe.devcluster.openshift.com:6443/api/v1/namespaces/kube-system?timeout=3s: context deadline exceeded (Client.Timeout exceeded while awaiting headers)
Jun 17 12:02:41.694 I openshift-apiserver OpenShift API started failing: Get https://api.qe-jian-0617.qe.devcluster.openshift.com:6443/apis/image.openshift.io/v1/namespaces/openshift-apiserver/imagestreams/missing?timeout=3s: context deadline exceeded (Client.Timeout exceeded while awaiting headers)
Jun 17 12:02:48.551 E kube-apiserver Kube API is not responding to GET requests
Jun 17 12:02:48.551 E openshift-apiserver OpenShift API is not responding to GET requests
Jun 17 12:02:52.132 I kube-apiserver Kube API started responding to GET requests
Jun 17 12:02:52.132 I openshift-apiserver OpenShift API started responding to GET requests
Jun 17 12:03:16.602 I ns/openshift-operator-lifecycle-manager pod/test-operator-f99jb node/ created
Jun 17 12:03:16.622 I ns/openshift-operator-lifecycle-manager pod/test-operator-f99jb Successfully assigned openshift-operator-lifecycle-manager/test-operator-f99jb to ip-10-0-141-71.us-east-2.compute.internal
Jun 17 12:03:24.245 I ns/openshift-operator-lifecycle-manager pod/test-operator-f99jb Pulling image "quay.io/dongboyan77/operator-registry:latest"
Jun 17 12:03:24.861 I ns/openshift-operator-lifecycle-manager pod/test-operator-f99jb Successfully pulled image "quay.io/dongboyan77/operator-registry:latest"
Jun 17 12:03:25.006 I ns/openshift-operator-lifecycle-manager pod/test-operator-f99jb Created container registry-server
Jun 17 12:03:25.035 I ns/openshift-operator-lifecycle-manager pod/test-operator-f99jb Started container registry-server
Jun 17 12:03:54.921 I ns/openshift-operators clusterserviceversion/etcdoperator.v0.9.2 requirements not yet checked
Jun 17 12:03:54.957 I ns/openshift-operators clusterserviceversion/etcdoperator.v0.9.2 one or more requirements couldn't be found
Jun 17 12:03:56.297 I ns/openshift-operators clusterserviceversion/etcdoperator.v0.9.2 all requirements found, attempting install
Jun 17 12:03:57.520 I ns/openshift-operators clusterserviceversion/etcdoperator.v0.9.2 all requirements found, attempting install (2 times)
Jun 17 12:03:58.337 I ns/openshift-operators clusterserviceversion/etcdoperator.v0.9.2 waiting for install components to report healthy
Jun 17 12:03:58.340 I ns/openshift-operators deployment/etcd-operator Scaled up replica set etcd-operator-c98b9f6bb to 1
Jun 17 12:03:58.341 I ns/openshift-operators pod/etcd-operator-c98b9f6bb-r794b node/ created
Jun 17 12:03:58.617 I ns/openshift-operators replicaset/etcd-operator-c98b9f6bb Created pod: etcd-operator-c98b9f6bb-r794b
Jun 17 12:03:58.620 I ns/openshift-operators pod/etcd-operator-c98b9f6bb-r794b Successfully assigned openshift-operators/etcd-operator-c98b9f6bb-r794b to ip-10-0-150-209.us-east-2.compute.internal
Jun 17 12:03:59.075 I ns/openshift-operators clusterserviceversion/etcdoperator.v0.9.2 waiting for install components to report healthy (2 times)
Jun 17 12:03:59.873 I ns/openshift-operators clusterserviceversion/etcdoperator.v0.9.2 installing: Waiting: waiting for deployment etcd-operator to become ready: Waiting for rollout to finish: 0 of 1 updated replicas are available...\n
Jun 17 12:04:06.190 I ns/openshift-operators pod/etcd-operator-c98b9f6bb-r794b Container image "quay.io/coreos/etcd-operator@sha256:c0301e4686c3ed4206e370b42de5a3bd2229b9fb4906cf85f3f30650424abec2" already present on machine
Jun 17 12:04:06.529 I ns/openshift-operators pod/etcd-operator-c98b9f6bb-r794b Created container etcd-operator
Jun 17 12:04:06.529 I ns/openshift-operators pod/etcd-operator-c98b9f6bb-r794b Started container etcd-operator
Jun 17 12:04:06.529 I ns/openshift-operators pod/etcd-operator-c98b9f6bb-r794b Container image "quay.io/coreos/etcd-operator@sha256:c0301e4686c3ed4206e370b42de5a3bd2229b9fb4906cf85f3f30650424abec2" already present on machine
Jun 17 12:04:06.603 I ns/openshift-operators pod/etcd-operator-c98b9f6bb-r794b Created container etcd-backup-operator
Jun 17 12:04:06.640 I ns/openshift-operators pod/etcd-operator-c98b9f6bb-r794b Started container etcd-backup-operator
Jun 17 12:04:06.645 I ns/openshift-operators pod/etcd-operator-c98b9f6bb-r794b Container image "quay.io/coreos/etcd-operator@sha256:c0301e4686c3ed4206e370b42de5a3bd2229b9fb4906cf85f3f30650424abec2" already present on machine
Jun 17 12:04:06.835 I ns/openshift-operators pod/etcd-operator-c98b9f6bb-r794b Created container etcd-restore-operator
Jun 17 12:04:06.839 I ns/openshift-operators pod/etcd-operator-c98b9f6bb-r794b Started container etcd-restore-operator
Jun 17 12:04:07.058 I ns/openshift-operators clusterserviceversion/etcdoperator.v0.9.2 install strategy completed with no errors
Jun 17 12:04:07.860 I ns/openshift-operators clusterserviceversion/etcdoperator.v0.9.2 install strategy completed with no errors (2 times)
Jun 17 12:04:17.894 I ns/openshift-machine-api machine/qe-jian-0617-fnm25-master-1 Updated machine qe-jian-0617-fnm25-master-1 (57 times)
Jun 17 12:04:18.943 I ns/openshift-machine-api machine/qe-jian-0617-fnm25-master-2 Updated machine qe-jian-0617-fnm25-master-2 (58 times)
Jun 17 12:04:19.112 I ns/openshift-machine-api machine/qe-jian-0617-fnm25-worker-us-east-2a-8lsrd Updated machine qe-jian-0617-fnm25-worker-us-east-2a-8lsrd (60 times)
Jun 17 12:04:19.329 I ns/openshift-machine-api machine/qe-jian-0617-fnm25-worker-us-east-2b-q42nk Updated machine qe-jian-0617-fnm25-worker-us-east-2b-q42nk (60 times)
Jun 17 12:04:20.374 I ns/openshift-machine-api machine/qe-jian-0617-fnm25-master-0 Updated machine qe-jian-0617-fnm25-master-0 (57 times)
Jun 17 12:04:23.814 I ns/openshift-operators endpoints/etcd-operator etcd-operator-c98b9f6bb-r794b became leader
Jun 17 12:04:24.025 I ns/openshift-operators endpoints/etcd-backup-operator etcd-operator-c98b9f6bb-r794b became leader
Jun 17 12:04:24.345 I ns/openshift-operators endpoints/etcd-restore-operator etcd-operator-c98b9f6bb-r794b became leader
Jun 17 12:05:05.838 I ns/openshift-operators pod/etcd-operator-c98b9f6bb-r794b Stopping container etcd-backup-operator
Jun 17 12:05:05.838 I ns/openshift-operators pod/etcd-operator-c98b9f6bb-r794b Stopping container etcd-operator
Jun 17 12:05:05.838 W ns/openshift-operators pod/etcd-operator-c98b9f6bb-r794b node/ip-10-0-150-209.us-east-2.compute.internal graceful deletion within 30s
Jun 17 12:05:05.842 I ns/openshift-operators pod/etcd-operator-c98b9f6bb-r794b Stopping container etcd-restore-operator
Jun 17 12:05:07.257 W ns/openshift-operators pod/etcd-operator-c98b9f6bb-r794b node/ip-10-0-150-209.us-east-2.compute.internal invariant violation (bug): pod should not transition Running->Pending even when terminated
Jun 17 12:05:07.257 W ns/openshift-operators pod/etcd-operator-c98b9f6bb-r794b node/ip-10-0-150-209.us-east-2.compute.internal container=etcd-backup-operator container stopped being ready
Jun 17 12:05:07.257 W ns/openshift-operators pod/etcd-operator-c98b9f6bb-r794b node/ip-10-0-150-209.us-east-2.compute.internal container=etcd-operator container stopped being ready
Jun 17 12:05:07.257 W ns/openshift-operators pod/etcd-operator-c98b9f6bb-r794b node/ip-10-0-150-209.us-east-2.compute.internal container=etcd-restore-operator container stopped being ready
Jun 17 12:05:08.101 W ns/openshift-operator-lifecycle-manager pod/test-operator-f99jb node/ip-10-0-141-71.us-east-2.compute.internal graceful deletion within 30s
Jun 17 12:05:08.478 I ns/openshift-operator-lifecycle-manager pod/test-operator-f99jb Stopping container registry-server
Jun 17 12:05:08.479 W ns/openshift-operators pod/etcd-operator-c98b9f6bb-r794b node/ip-10-0-150-209.us-east-2.compute.internal deleted
Jun 17 12:05:08.977 E ns/openshift-operator-lifecycle-manager pod/test-operator-f99jb node/ip-10-0-141-71.us-east-2.compute.internal container=registry-server container exited with code 2 (Error): 
Jun 17 12:05:14.028 W ns/openshift-operator-lifecycle-manager pod/test-operator-f99jb node/ip-10-0-141-71.us-east-2.compute.internal deleted

Writing JUnit report to junit_e2e_20190617-120517.xml

7 pass, 0 skip (2m44s)
```

jianzhangbjz commented:
/test e2e-aws

jianzhangbjz commented:
Does anyone know how to fix below errors for e2e-aws test?
```
Jun 18 02:51:41.289 I ns/openshift-operator-lifecycle-manager pod/test-operator-lfrc6 Started container registry-server
Jun 18 02:51:46.182 W persistentvolume/pvc-f0927cc2-9173-11e9-ab71-0a6266c299e2 Error deleting EBS volume "vol-0c64efc6bf329d029" since volume is currently attached to "i-06f7b3c66d769a4c8"
Jun 18 02:52:36.787 W persistentvolume/pvc-de87485a-9173-11e9-ab71-0a6266c299e2 Error deleting EBS volume "vol-01459fcc95446ae88" since volume is currently attached to "i-06f7b3c66d769a4c8"
Jun 18 02:52:36.983 W persistentvolume/pvc-f730f0df-9173-11e9-ab71-0a6266c299e2 Error deleting EBS volume "vol-094978438bfb4843b" since volume is currently attached to "i-024d5a2c321da92a6"
Jun 18 02:52:42.572 W persistentvolume/pvc-087aa4c8-9174-11e9-924d-1253a42b69e8 Error deleting EBS volume "vol-021e728a1dba53f76" since volume is currently attached to "i-06f7b3c66d769a4c8"
Jun 18 02:52:58.185 W persistentvolume/pvc-2c7c9752-9174-11e9-b59e-129d76895528 Error deleting EBS volume "vol-055a85104530fc227" since volume is in "creating" state
```

jianzhangbjz commented:
/retest

jianzhangbjz commented:
@robszumski Could you help review our automation test case? Since it is our first one for Origin, do you have any suggestions about the test case structure? Thanks for your time!

jianzhangbjz commented:
/retest

jianzhangbjz commented:
Not sure why always got below errors. Actually, this scenario works well in my local OCP 4.1. @robszumski Do you know how to fix this error or know some guys could help take a look at it?
```console
p-10-0-157-79.ec2.internal deleted
Jun 19 01:34:33.260 W persistentvolume/pvc-3db145b2-9232-11e9-bfcf-12593d092248 Error deleting EBS volume "vol-0ba9372d182cc3b99" since volume is currently attached to "i-0e599ac309017d42c"
Jun 19 01:35:54.744 W persistentvolume/pvc-d3b1ecbb-9231-11e9-bfcf-12593d092248 Error deleting EBS volume "vol-0997765599c0a2eb6" since volume is currently attached to "i-0168df796fbec877d"
Jun 19 01:35:54.782 W persistentvolume/pvc-e3f97e9b-9231-11e9-bfcf-12593d092248 Error deleting EBS volume "vol-03f2a0833e4ec64ac" since volume is currently attached to "i-0aede5103452a601a"
Jun 19 01:35:54.834 W persistentvolume/pvc-f51fbbb2-9231-11e9-bfcf-12593d092248 Error deleting EBS volume "vol-09e9fa6c83010d9bb" since volume is currently attached to "i-0e599ac309017d42c"
Jun 19 01:37:42.864 I ns/openshift-machine-api machine/ci-op-x1j6jht0-55c01-pff2b-worker-us-east-1b-8tgj6 Updated machine ci-op-x1j6jht0-55c01-pff2b-worker-us-east-1b-8tgj6 (10 times)
Jun 19 01:37:44.777 I ns/openshift-machine-api machine/ci-op-x1j6jht0-55c01-pff2b-master-2 Updated machine ci-op-x1j6jht0-55c01-pff2b-master-2 (8 times)
Jun 19 01:37:46.500 I ns/openshift-machine-api machine/ci-op-x1j6jht0-55c01-pff2b-master-0 Updated machine ci-op-x1j6jht0-55c01-pff2b-master-0 (8 times)
Jun 19 01:37:47.855 I ns/openshift-machine-api machine/ci-op-x1j6jht0-55c01-pff2b-master-1 Updated machine ci-op-x1j6jht0-55c01-pff2b-master-1 (8 times)
Jun 19 01:37:48.098 I ns/openshift-machine-api machine/ci-op-x1j6jht0-55c01-pff2b-worker-us-east-1a-2kcj9 Updated machine ci-op-x1j6jht0-55c01-pff2b-worker-us-east-1a-2kcj9 (10 times)
Jun 19 01:37:48.292 I ns/openshift-machine-api machine/ci-op-x1j6jht0-55c01-pff2b-worker-us-east-1a-q2rgd Updated machine ci-op-x1j6jht0-55c01-pff2b-worker-us-east-1a-q2rgd (10 times)

Failing tests:

[Feature:Platform] OLM should [ocp-22070]support grpc sourcetype [Suite:openshift/conformance/parallel]

Writing JUnit report to /tmp/artifacts/junit/junit_e2e_20190619-013912.xml

error: 1 fail, 822 pass, 1265 skip (34m46s)
```

bparees commented:
/assign @ecordell 

@ecordell can you or someone on your team review this?

jianzhangbjz commented:
@ecordell Could you help review this PR? Thanks!

njhale commented on test/extended/operators/olm.go:
```diff
@@ -37,12 +36,72 @@ var _ = g.Describe("[Feature:Platform] OLM should", func() {
+		createResourceByYAML(oc, subscriptionYF)
+		time.Sleep(60 * time.Second)
+		getCmds := map[int][]string{
+			0: {"get", "sub", "-n", "openshift-operators", "test-operator"},
```
Is there any reason for this to be a map?

ecordell replied:
I assume so that order is enforced; go slices are iterated randomly

jianzhangbjz replied:
Yes, thanks! I'd like to check them in order.

jianzhangbjz replied:
@njhale Since `checkResource` receive the `slice` as arguments, I intended to put these `slices` into together. So I used `map`. I will sort this `map`, does it make sense? Or any suggestions? Thanks!

ecordell replied:
@jianzhangbjz You can have a slice of slices, e.g.:

```go 
getCmds := [][]string{
  {"get", "sub", "-n", "openshift-operators", "test-operator"},
  {"get", "csv", "-n", "openshift-operators", "etcdoperator.v0.9.2"},
  {"get", "pods", "-n", "openshift-operators", "etcd-operator"},
}
for _, c := range getCmds {
  checkResource(oc, c)
}
```

njhale replied:
@jianzhangbjz adding to @ecordell 's suggestion, you can also create a struct to hold the arguments:

```go
type resource struct {
    kind string
    namespace string
    name string
}

get := []resource{
    {"sub", "openshift-operators", "test-operator"},
    {"csv", "openshift-operators", "etcdoperator.v0.9.2"},
    {"pods", "openshift-operators", "etcd-operator"},
}

// ...
func checkResource(oc *exutil.CLI, res resource) {
    output, err := oc.AsAdmin().WithoutNamespace().Run("get").Args(fmt.Sprintf("%s -n %s %s", res.kind, res.namespace, res.name)).Output()
    // ...    
}
```

jianzhangbjz commented on test/extended/operators/olm.go:
```diff
@@ -37,12 +36,72 @@ var _ = g.Describe("[Feature:Platform] OLM should", func() {
+	// OCP-22070
+
+	g.It("[ocp-22070]support grpc sourcetype", func() {
+		catalogsourceYF := "https://raw.githubusercontent.com/jianzhangbjz/v3-testfiles/v4.1/olm/catalogsource/image-catalogsource.yaml"
```
Yes, thanks! For me, I don't like to use this kind of way to implement it. But, I could not find a better solution. What do you suggest? 
I will change this file repo to official one once we decide to use this solution.

jianzhangbjz replied:
@bparees @njhale Is it available if we add these YAML files under a newly created folder in this Origin repo?

bparees replied:
@jianzhangbjz yes, we use bindata for this in our other tests.  e.g.:

https://github.com/openshift/origin/tree/master/test/testdata
https://github.com/openshift/origin/blob/master/test/extended/builds/build_pruning.go#L27

you'll have to run hack/update-generated-bindata.sh if you introduce new fixture content.

bparees replied:
(as I recall our test framework extracts the files from the bindata.go and makes them available on the filesystem when the tests execute)

jianzhangbjz replied:
@bparees Sounds good! Thanks! I will have a try!

jianzhangbjz replied:
@bparees How can we set up these YAML files to receive parameter when running test?

bparees replied:
make them openshift templates and then instantiate them?

jianzhangbjz replied:
Yes, do you have any suggestions?

bparees replied:
i'm confused about the question.  You should wrap the catalogsource object in a template and parameterize it, and then use `oc process` from within your test to instantiate it with whatever parameters you need.

jianzhangbjz replied:
Yes, thanks! One more thing. I guess it's inappropriate to submit this PR to `master` branch since it's always changing. I should submit it to `release-4.1` branch since it only covers 4.1 scenarios, right?

bparees replied:
i haven't looked at exactly what this test does that would make it 4.1 specific, but it's unusual to have branch-specific test cases.  Why is it not applicable to 4.2/master?

jianzhangbjz commented:
> I have a few concerns with this PR, but the biggest one is that it externalizes/duplicates testing that is already done by OLM's e2e tests.

@njhale Thanks! But, I guess those exist e2e tests in OLM repo, not the Origin. Adding these tests in the `test/extended` folder of Origin is to cover different downstream OCP metrics automatically. I also don't like to do duplicate work, correct me if I'm wrong. Many thanks!

jianzhangbjz commented:
/retest

jianzhangbjz commented:
/retest

ecordell commented on test/extended/operators/olm.go:
```diff
@@ -70,4 +72,69 @@ var _ = g.Describe("[Feature:Platform] OLM should", func() {
+		checkResource(oc, catalogCmd)
+
+		createResourceByYAML(oc, subscriptionConfig)
+		time.Sleep(60 * time.Second)
```
Can we remove this sleep by polling or `watch`ing a resource instead?

jianzhangbjz replied:
@ecordell Yes, thanks, updated it.

ecordell commented on test/extended/operators/olm.go:
```diff
@@ -70,4 +72,69 @@ var _ = g.Describe("[Feature:Platform] OLM should", func() {
+		subscriptionConfig := filepath.Join(buildPruningBaseDir, "image-sub.yaml")
+
+		createResourceByYAML(oc, catalogsourceConfig)
+		time.Sleep(30 * time.Second)
```
Can we remove this sleep by polling or `watch`ing a resource instead?

jianzhangbjz replied:
Yes, thanks, updated it.

jianzhangbjz commented:
/test e2e-aws

jianzhangbjz commented:
/test e2e-aws

jianzhangbjz commented:
/retest

jianzhangbjz commented:
/test unit
/test e2e-aws

njhale commented on test/extended/operators/olm.go:
```diff
@@ -125,16 +125,19 @@ func createResourceByYAML(oc *exutil.CLI, yaml string) {
-		e2e.Failf("Unable to get the output:%v", commands)
-	}
+	err := wait.Poll(3*time.Second, 60*time.Second, func() (bool, error) {
+		output, err := oc.AsAdmin().WithoutNamespace().Run(commands[0]).Args(commands[1 : len(commands)-1]...).Output()
```
nit: `commands[1:]` is the same as `commands[1:len(commands)-1]`

jianzhangbjz replied:
Ah, yes, thank you!

jianzhangbjz commented:
@njhale Do you know how to solve the errors below?
```console
fail [github.com/openshift/origin/test/extended/operators/olm.go:143]: Unexpected error:
    <*util.ExitError | 0xc002794720>: {
        Cmd: "oc --config=/tmp/admin.kubeconfig get packagemanifest",
        StdErr: "Error from server: no table handler registered for this type *operators.PackageManifestList",
        ExitError: {
            ProcessState: {
                pid: 6440,
                status: 256,
                rusage: {
                    Utime: {Sec: 0, Usec: 233140},
                    Stime: {Sec: 0, Usec: 61895},
                    Maxrss: 122336,
                    Ixrss: 0,
                    Idrss: 0,
                    Isrss: 0,
                    Minflt: 11536,
                    Majflt: 0,
                    Nswap: 0,
                    Inblock: 0,
                    Oublock: 0,
                    Msgsnd: 0,
                    Msgrcv: 0,
                    Nsignals: 0,
                    Nvcsw: 831,
                    Nivcsw: 16,
                },
            },
            Stderr: nil,
        },
    }
    exit status 1
occurred
```

jianzhangbjz commented:
/retest

jianzhangbjz commented:
/retest

jianzhangbjz commented:
/test e2e-gcp

jianzhangbjz commented:
/retest

jianzhangbjz commented:
cc: @tbuskey

jianzhangbjz commented:
Running logs in my laptop:
```console
Cluster version is 4.3.0-0.nightly-2019-11-28-190856
mac:openshift-tests jianzhang$ ./openshift-tests run all --dry-run | grep "\[Feature:Platform\] OLM should"  | ./openshift-tests run --junit-dir=./ -f -
Nov 29 16:42:16.131: INFO: >>> kubeConfig: /Users/jianzhang/43-kubeconfig
Nov 29 16:42:16.133: INFO: >>> kubeConfig: /Users/jianzhang/43-kubeconfig
started: (0/1/8) "[Feature:Platform] OLM should [ocp-22070]support grpc sourcetype [Suite:openshift/conformance/parallel]"

started: (0/2/8) "[Feature:Platform] OLM should be installed with packagemanifests at version v1 [Suite:openshift/conformance/parallel]"

started: (0/3/8) "[Feature:Platform] OLM should be installed with installplans at version v1alpha1 [Suite:openshift/conformance/parallel]"

started: (0/4/8) "[Feature:Platform] OLM should be installed with clusterserviceversions at version v1alpha1 [Suite:openshift/conformance/parallel]"

started: (0/5/8) "[Feature:Platform] OLM should be installed with operatorgroups at version v1 [Suite:openshift/conformance/parallel]"

started: (0/6/8) "[Feature:Platform] OLM should have imagePullPolicy:IfNotPresent on thier deployments [Suite:openshift/conformance/parallel]"

started: (0/7/8) "[Feature:Platform] OLM should be installed with catalogsources at version v1alpha1 [Suite:openshift/conformance/parallel]"

started: (0/8/8) "[Feature:Platform] OLM should be installed with subscriptions at version v1alpha1 [Suite:openshift/conformance/parallel]"

passed: (26s) 2019-11-29T08:42:50 "[Feature:Platform] OLM should be installed with packagemanifests at version v1 [Suite:openshift/conformance/parallel]"

passed: (26.2s) 2019-11-29T08:42:50 "[Feature:Platform] OLM should be installed with operatorgroups at version v1 [Suite:openshift/conformance/parallel]"

passed: (26.2s) 2019-11-29T08:42:50 "[Feature:Platform] OLM should be installed with catalogsources at version v1alpha1 [Suite:openshift/conformance/parallel]"

passed: (26.6s) 2019-11-29T08:42:50 "[Feature:Platform] OLM should be installed with subscriptions at version v1alpha1 [Suite:openshift/conformance/parallel]"

passed: (27.1s) 2019-11-29T08:42:51 "[Feature:Platform] OLM should be installed with clusterserviceversions at version v1alpha1 [Suite:openshift/conformance/parallel]"

passed: (28.3s) 2019-11-29T08:42:52 "[Feature:Platform] OLM should have imagePullPolicy:IfNotPresent on thier deployments [Suite:openshift/conformance/parallel]"

passed: (29.6s) 2019-11-29T08:42:53 "[Feature:Platform] OLM should be installed with installplans at version v1alpha1 [Suite:openshift/conformance/parallel]"

passed: (1m22s) 2019-11-29T08:43:46 "[Feature:Platform] OLM should [ocp-22070]support grpc sourcetype [Suite:openshift/conformance/parallel]"


Timeline:

Nov 29 08:42:49.791 I ns/openshift-marketplace pod/test-operator-54npg node/ created
Nov 29 08:42:49.890 I ns/openshift-marketplace pod/test-operator-54npg Successfully assigned openshift-marketplace/test-operator-54npg to ip-10-0-151-251.us-east-2.compute.internal
Nov 29 08:42:50.026 W ns/openshift-marketplace pod/test-operator-54npg node/ip-10-0-151-251.us-east-2.compute.internal graceful deletion within 0s
Nov 29 08:42:50.034 W ns/openshift-marketplace pod/test-operator-54npg node/ip-10-0-151-251.us-east-2.compute.internal deleted
Nov 29 08:42:50.129 I ns/openshift-marketplace pod/test-operator-qwm7r node/ created
Nov 29 08:42:50.235 I ns/openshift-marketplace pod/test-operator-qwm7r Successfully assigned openshift-marketplace/test-operator-qwm7r to ip-10-0-151-251.us-east-2.compute.internal
Nov 29 08:42:57.239 I ns/openshift-marketplace pod/test-operator-qwm7r Pulling image "quay.io/aosqe42/operator-registry:auto"
Nov 29 08:42:59.716 I ns/openshift-marketplace pod/test-operator-qwm7r Successfully pulled image "quay.io/aosqe42/operator-registry:auto"
Nov 29 08:42:59.891 I ns/openshift-marketplace pod/test-operator-qwm7r Created container registry-server
Nov 29 08:42:59.901 I ns/openshift-marketplace pod/test-operator-qwm7r Started container registry-server
Nov 29 08:43:28.864 I ns/openshift-operators clusterserviceversion/etcdoperator.v0.9.2 requirements not yet checked
Nov 29 08:43:32.659 I ns/openshift-operators clusterserviceversion/etcdoperator.v0.9.2 all requirements found, attempting install
Nov 29 08:43:38.290 I ns/openshift-operators clusterserviceversion/etcdoperator.v0.9.2 waiting for install components to report healthy
Nov 29 08:43:38.290 I ns/openshift-operators deployment/etcd-operator Scaled up replica set etcd-operator-5b685b89f6 to 1
Nov 29 08:43:38.291 I ns/openshift-operators pod/etcd-operator-5b685b89f6-s8pr8 node/ created
Nov 29 08:43:38.292 I ns/openshift-operators replicaset/etcd-operator-5b685b89f6 Created pod: etcd-operator-5b685b89f6-s8pr8
Nov 29 08:43:38.528 I ns/openshift-operators pod/etcd-operator-5b685b89f6-s8pr8 Successfully assigned openshift-operators/etcd-operator-5b685b89f6-s8pr8 to ip-10-0-154-130.us-east-2.compute.internal
Nov 29 08:43:39.462 I ns/openshift-operators clusterserviceversion/etcdoperator.v0.9.2 installing: Waiting: waiting for deployment etcd-operator to become ready: Waiting for rollout to finish: 0 of 1 updated replicas are available...\n
Nov 29 08:43:41.080 W ns/openshift-marketplace pod/test-operator-qwm7r node/ip-10-0-151-251.us-east-2.compute.internal graceful deletion within 30s
Nov 29 08:43:41.092 I ns/openshift-marketplace pod/test-operator-qwm7r Stopping container registry-server
Nov 29 08:43:42.079 E ns/openshift-marketplace pod/test-operator-qwm7r node/ip-10-0-151-251.us-east-2.compute.internal container=registry-server container exited with code 2 (Error): 
Nov 29 08:43:45.530 W ns/openshift-operators pod/etcd-operator-5b685b89f6-s8pr8 node/ip-10-0-154-130.us-east-2.compute.internal graceful deletion within 30s
Nov 29 08:43:45.543 E ns/openshift-operators pod/etcd-operator-5b685b89f6-s8pr8 node/ip-10-0-154-130.us-east-2.compute.internal container=etcd-backup-operator container exited with code 137 (ContainerStatusUnknown): The container could not be located when the pod was terminated
Nov 29 08:43:45.543 E ns/openshift-operators pod/etcd-operator-5b685b89f6-s8pr8 node/ip-10-0-154-130.us-east-2.compute.internal container=etcd-operator container exited with code 137 (ContainerStatusUnknown): The container could not be located when the pod was terminated
Nov 29 08:43:45.543 E ns/openshift-operators pod/etcd-operator-5b685b89f6-s8pr8 node/ip-10-0-154-130.us-east-2.compute.internal container=etcd-restore-operator container exited with code 137 (ContainerStatusUnknown): The container could not be located when the pod was terminated
Nov 29 08:43:45.892 I ns/openshift-operators pod/etcd-operator-5b685b89f6-s8pr8 Container image "quay.io/coreos/etcd-operator@sha256:c0301e4686c3ed4206e370b42de5a3bd2229b9fb4906cf85f3f30650424abec2" already present on machine
Nov 29 08:43:45.897 W ns/openshift-operators pod/etcd-operator-5b685b89f6-s8pr8 Error: cannot find volume "etcd-operator-token-m2xlt" to mount into container "etcd-operator"
Nov 29 08:43:46.017 I ns/openshift-operators pod/etcd-operator-5b685b89f6-s8pr8 Container image "quay.io/coreos/etcd-operator@sha256:c0301e4686c3ed4206e370b42de5a3bd2229b9fb4906cf85f3f30650424abec2" already present on machine
Nov 29 08:43:46.017 W ns/openshift-operators pod/etcd-operator-5b685b89f6-s8pr8 Error: cannot find volume "etcd-operator-token-m2xlt" to mount into container "etcd-backup-operator"
Nov 29 08:43:46.121 I ns/openshift-operators pod/etcd-operator-5b685b89f6-s8pr8 Container image "quay.io/coreos/etcd-operator@sha256:c0301e4686c3ed4206e370b42de5a3bd2229b9fb4906cf85f3f30650424abec2" already present on machine
Nov 29 08:43:46.121 W ns/openshift-operators pod/etcd-operator-5b685b89f6-s8pr8 Error: cannot find volume "etcd-operator-token-m2xlt" to mount into container "etcd-restore-operator"
Nov 29 08:43:46.121 I ns/openshift-operators clusterserviceversion/etcdoperator.v0.9.2 installing: ComponentMissing: missing deployment with name=etcd-operator

Writing JUnit report to junit_e2e_20191129-084346.xml

8 pass, 0 skip (1m22s)

```

jianzhangbjz commented:
/retest

jianzhangbjz commented:
@shawn-hurley @bparees Could you help add the approval labels? Thanks!

jianzhangbjz commented:
@emmajiafan Could you have a check for below errors? Thanks!
```console
[Feature:Marketplace] Marketplace resources with labels provider displayName [ocp-21728]create opsrc with labels [Suite:openshift/conformance/parallel] expand_less	47s
fail [github.com/openshift/origin/test/extended/marketplace/marketplace_lables.go:120]: Expected
    <string>: 
to contain substring
    <string>: camel-k-marketplace-e2e-tests
```

jianzhangbjz commented:
/test e2e-gcp

jianzhangbjz commented:
I test this `Marketplace` scenarios, it does pass in my laptop, see below. Anything wrong in our CI cluster?
```console
mac:openshift-tests jianzhang$ oc get clusterversion
NAME      VERSION                             AVAILABLE   PROGRESSING   SINCE   STATUS
version   4.3.0-0.nightly-2019-11-29-051144   True        False         47m     Cluster version is 4.3.0-0.nightly-2019-11-29-051144
mac:openshift-tests jianzhang$ ./openshift-tests run all --dry-run | grep "\[Feature:Marketplace\] Marketplace resources with labels provider displayName"  | ./openshift-tests run --junit-dir=./ -f -
Dec  2 10:41:10.283: INFO: >>> kubeConfig: /Users/jianzhang/43-kubeconfig
Dec  2 10:41:10.286: INFO: >>> kubeConfig: /Users/jianzhang/43-kubeconfig
started: (0/1/1) "[Feature:Marketplace] Marketplace resources with labels provider displayName [ocp-21728]create opsrc with labels [Suite:openshift/conformance/parallel]"

Dec  2 10:41:23.890: INFO: >>> kubeConfig: /Users/jianzhang/43-kubeconfig
Dec  2 10:41:23.892: INFO: >>> kubeConfig: /Users/jianzhang/43-kubeconfig
Dec  2 10:41:29.920: INFO: >>> kubeConfig: /Users/jianzhang/43-kubeconfig
Dec  2 10:41:29.922: INFO: Waiting up to 30m0s for all (but 100) nodes to be schedulable
Dec  2 10:41:30.746: INFO: Waiting up to 10m0s for all pods (need at least 0) in namespace 'kube-system' to be running and ready
Dec  2 10:41:31.609: INFO: 0 / 0 pods in namespace 'kube-system' are running and ready (0 seconds elapsed)
Dec  2 10:41:31.609: INFO: expected 0 pod replicas in namespace 'kube-system', 0 are Running and Ready.
Dec  2 10:41:31.609: INFO: Waiting up to 5m0s for all daemonsets in namespace 'kube-system' to start
Dec  2 10:41:31.938: INFO: e2e test version: v0.0.0-master+$Format:%h$
Dec  2 10:41:32.242: INFO: kube-apiserver version: v1.16.2
Dec  2 10:41:32.242: INFO: >>> kubeConfig: /Users/jianzhang/43-kubeconfig
Dec  2 10:41:32.551: INFO: Cluster IP family: ipv4
[BeforeEach] [Top Level]
  /Users/jianzhang/goproject/src/github.com/openshift/origin/test/extended/util/test.go:60
[BeforeEach] [Feature:Marketplace] Marketplace resources with labels provider displayName
  /Users/jianzhang/goproject/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/test/e2e/framework/framework.go:151
STEP: Creating a kubernetes client
Dec  2 10:41:32.556: INFO: >>> kubeConfig: /Users/jianzhang/43-kubeconfig
[BeforeEach] [Feature:Marketplace] Marketplace resources with labels provider displayName
  /Users/jianzhang/goproject/src/github.com/openshift/origin/test/extended/util/client.go:109
Dec  2 10:41:35.525: INFO: configPath is now "/var/folders/2c/4whhm34n7892mf9l2j02cf480000gn/T/configfile252555297"
Dec  2 10:41:35.525: INFO: The user is now "e2e-test-marketplace-8lnlw-user"
Dec  2 10:41:35.525: INFO: Creating project "e2e-test-marketplace-8lnlw"
Dec  2 10:41:36.032: INFO: Waiting on permissions in project "e2e-test-marketplace-8lnlw" ...
Dec  2 10:41:36.285: INFO: Waiting for ServiceAccount "default" to be provisioned...
Dec  2 10:41:36.647: INFO: Waiting for ServiceAccount "deployer" to be provisioned...
Dec  2 10:41:37.002: INFO: Waiting for ServiceAccount "builder" to be provisioned...
Dec  2 10:41:37.464: INFO: Waiting for RoleBinding "system:image-pullers" to be provisioned...
Dec  2 10:41:37.976: INFO: Waiting for RoleBinding "system:image-builders" to be provisioned...
Dec  2 10:41:38.590: INFO: Waiting for RoleBinding "system:deployers" to be provisioned...
Dec  2 10:41:39.093: INFO: Project "e2e-test-marketplace-8lnlw" has been fully provisioned.
[It] [ocp-21728]create opsrc with labels [Suite:openshift/conformance/parallel]
  /Users/jianzhang/goproject/src/github.com/openshift/origin/test/extended/marketplace/marketplace_lables.go:44
Dec  2 10:41:39.093: INFO: Running 'oc --namespace=e2e-test-marketplace-8lnlw --config=/Users/jianzhang/43-kubeconfig process --ignore-unknown-parameters=true -f /var/folders/2c/4whhm34n7892mf9l2j02cf480000gn/T/fixture-testdata-dir566895946/test/extended/testdata/marketplace/opsrc/02-opsrc.yaml -p NAME=opsrctestlabel NAMESPACE=marketplace_e2e LABEL=optestlabel DISPLAYNAME=optestlabel PUBLISHER=optestlabel MARKETPLACE=openshift-marketplace'
Dec  2 10:41:40.449: INFO: Start to create Resource: /tmp/e2e-test-marketplace-8lnlw-config.json
Dec  2 10:41:40.449: INFO: Running 'oc --config=/Users/jianzhang/43-kubeconfig create -f /tmp/e2e-test-marketplace-8lnlw-config.json'
operatorsource.operators.coreos.com/opsrctestlabel created
Dec  2 10:41:46.571: INFO: Running 'oc --namespace=e2e-test-marketplace-8lnlw --config=/Users/jianzhang/43-kubeconfig get operatorsource opsrctestlabel -o=jsonpath={.status.currentPhase.phase.message} -n openshift-marketplace'
Dec  2 10:41:47.912: INFO: Running 'oc --config=/Users/jianzhang/43-kubeconfig get -n openshift-marketplace operatorsource opsrctestlabel -o=jsonpath={.metadata.labels.opsrc-provider}'
Dec  2 10:41:49.244: INFO: Running 'oc --config=/Users/jianzhang/43-kubeconfig get -n openshift-marketplace operatorsource opsrctestlabel -o=jsonpath={.spec.displayName}'
Dec  2 10:41:50.491: INFO: Running 'oc --config=/Users/jianzhang/43-kubeconfig get -n openshift-marketplace operatorsource opsrctestlabel -o=jsonpath={.spec.publisher}'
Dec  2 10:41:52.501: INFO: Running 'oc --config=/Users/jianzhang/43-kubeconfig get -n openshift-marketplace catalogsource opsrctestlabel -o=jsonpath={.metadata.labels.opsrc-provider}'
Dec  2 10:41:53.792: INFO: Running 'oc --config=/Users/jianzhang/43-kubeconfig get -n openshift-marketplace catalogsource opsrctestlabel -o=jsonpath={.spec.displayName}'
Dec  2 10:41:54.978: INFO: Running 'oc --config=/Users/jianzhang/43-kubeconfig get -n openshift-marketplace catalogsource opsrctestlabel -o=jsonpath={.spec.publisher}'
Dec  2 10:41:56.085: INFO: Running 'oc --namespace=e2e-test-marketplace-8lnlw --config=/Users/jianzhang/43-kubeconfig process --ignore-unknown-parameters=true -f /var/folders/2c/4whhm34n7892mf9l2j02cf480000gn/T/fixture-testdata-dir566895946/test/extended/testdata/marketplace/csc/02-csc.yaml -p NAME=csctestlabel NAMESPACE=openshift-operators MARKETPLACE=openshift-marketplace PACKAGES=camel-k-marketplace-e2e-tests DISPLAYNAME=csctestlabel PUBLISHER=csctestlabel'
Dec  2 10:41:57.579: INFO: Start to create Resource: /tmp/e2e-test-marketplace-8lnlw-config.json
Dec  2 10:41:57.579: INFO: Running 'oc --config=/Users/jianzhang/43-kubeconfig create -f /tmp/e2e-test-marketplace-8lnlw-config.json'
catalogsourceconfig.operators.coreos.com/csctestlabel created
Dec  2 10:42:03.748: INFO: Running 'oc --namespace=e2e-test-marketplace-8lnlw --config=/Users/jianzhang/43-kubeconfig get catalogsourceconfig csctestlabel -o=jsonpath={.status.currentPhase.phase.message} -n openshift-marketplace'
Dec  2 10:42:05.015: INFO: Running 'oc --config=/Users/jianzhang/43-kubeconfig get -n openshift-marketplace catalogsourceconfig csctestlabel -o=jsonpath={.spec.csDisplayName}'
Dec  2 10:42:06.377: INFO: Running 'oc --config=/Users/jianzhang/43-kubeconfig get -n openshift-marketplace catalogsourceconfig csctestlabel -o=jsonpath={.spec.csPublisher}'
Dec  2 10:42:07.460: INFO: Running 'oc --config=/Users/jianzhang/43-kubeconfig get -n openshift-operators catalogsource csctestlabel -o=jsonpath={.spec.displayName}'
Dec  2 10:42:08.690: INFO: Running 'oc --config=/Users/jianzhang/43-kubeconfig get -n openshift-operators catalogsource csctestlabel -o=jsonpath={.spec.publisher}'
Dec  2 10:42:09.760: INFO: Running 'oc --config=/Users/jianzhang/43-kubeconfig get -n openshift-marketplace operatorsource opsrctestlabel -o=jsonpath={.status.packages}'
Dec  2 10:42:10.885: INFO: Running 'oc --config=/Users/jianzhang/43-kubeconfig get packagemanifests -lopsrc-provider=optestlabel -o=name -n openshift-marketplace'
[AfterEach] [Feature:Marketplace] Marketplace resources with labels provider displayName
  /Users/jianzhang/goproject/src/github.com/openshift/origin/test/extended/util/client.go:101
Dec  2 10:42:13.694: INFO: Deleted {user.openshift.io/v1, Resource=users  e2e-test-marketplace-8lnlw-user}, err: <nil>
Dec  2 10:42:14.020: INFO: Deleted {oauth.openshift.io/v1, Resource=oauthclients  e2e-client-e2e-test-marketplace-8lnlw}, err: <nil>
Dec  2 10:42:14.302: INFO: Deleted {oauth.openshift.io/v1, Resource=oauthaccesstokens  NdDo_oGMQA-dvAvzKhyEZgAAAAAAAAAA}, err: <nil>
[AfterEach] [Feature:Marketplace] Marketplace resources with labels provider displayName
  /Users/jianzhang/goproject/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/test/e2e/framework/framework.go:152
Dec  2 10:42:14.302: INFO: Waiting up to 7m0s for all (but 100) nodes to be ready
STEP: Destroying namespace "e2e-test-marketplace-8lnlw" for this suite.
[AfterEach] [Feature:Marketplace] Marketplace resources with labels provider displayName
  /Users/jianzhang/goproject/src/github.com/openshift/origin/test/extended/marketplace/marketplace_lables.go:30
Dec  2 10:42:15.248: INFO: Running 'oc --config=/Users/jianzhang/43-kubeconfig delete -n openshift-marketplace operatorsource opsrctestlabel'
Dec  2 10:42:16.862: INFO: Running 'oc --config=/Users/jianzhang/43-kubeconfig delete -n openshift-marketplace catalogsourceconfig csctestlabel'
Dec  2 10:42:18.224: INFO: Running AfterSuite actions on all nodes
Dec  2 10:42:18.224: INFO: Running AfterSuite actions on node 1

passed: (55.6s) 2019-12-02T02:42:18 "[Feature:Marketplace] Marketplace resources with labels provider displayName [ocp-21728]create opsrc with labels [Suite:openshift/conformance/parallel]"


Timeline:

Dec 02 02:41:41.792 I ns/openshift-marketplace deployment/opsrctestlabel Scaled up replica set opsrctestlabel-7784d95d48 to 1
Dec 02 02:41:41.839 I ns/openshift-marketplace pod/opsrctestlabel-7784d95d48-94m2j node/ created
Dec 02 02:41:41.842 I ns/openshift-marketplace replicaset/opsrctestlabel-7784d95d48 Created pod: opsrctestlabel-7784d95d48-94m2j
Dec 02 02:41:42.037 I ns/openshift-marketplace pod/opsrctestlabel-7784d95d48-94m2j Successfully assigned openshift-marketplace/opsrctestlabel-7784d95d48-94m2j to ip-10-0-159-84.us-east-2.compute.internal
Dec 02 02:41:50.167 I ns/openshift-marketplace pod/opsrctestlabel-7784d95d48-94m2j Pulling image "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:9fd355d207a75d6944223c315e3dee803fbacf071912359909f0d443a5e6efec"
Dec 02 02:41:58.826 I ns/openshift-marketplace deployment/csctestlabel Scaled up replica set csctestlabel-cb548bf6f to 1
Dec 02 02:41:58.856 I ns/openshift-marketplace pod/csctestlabel-cb548bf6f-xp9qg node/ created
Dec 02 02:41:58.870 I ns/openshift-marketplace replicaset/csctestlabel-cb548bf6f Created pod: csctestlabel-cb548bf6f-xp9qg
Dec 02 02:41:58.896 I ns/openshift-marketplace pod/csctestlabel-cb548bf6f-xp9qg Successfully assigned openshift-marketplace/csctestlabel-cb548bf6f-xp9qg to ip-10-0-159-84.us-east-2.compute.internal
Dec 02 02:41:59.102 I ns/openshift-marketplace deployment/csctestlabel Scaled up replica set csctestlabel-674c7856bb to 1
Dec 02 02:41:59.103 I ns/openshift-marketplace pod/csctestlabel-674c7856bb-z42qw node/ created
Dec 02 02:41:59.117 I ns/openshift-marketplace replicaset/csctestlabel-674c7856bb Created pod: csctestlabel-674c7856bb-z42qw
Dec 02 02:41:59.117 I ns/openshift-marketplace pod/csctestlabel-674c7856bb-z42qw Successfully assigned openshift-marketplace/csctestlabel-674c7856bb-z42qw to ip-10-0-131-240.us-east-2.compute.internal
Dec 02 02:41:59.196 I ns/openshift-marketplace pod/opsrctestlabel-7784d95d48-94m2j Successfully pulled image "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:9fd355d207a75d6944223c315e3dee803fbacf071912359909f0d443a5e6efec"
Dec 02 02:41:59.649 I ns/openshift-marketplace pod/opsrctestlabel-7784d95d48-94m2j Created container opsrctestlabel
Dec 02 02:41:59.718 I ns/openshift-marketplace pod/opsrctestlabel-7784d95d48-94m2j Started container opsrctestlabel
Dec 02 02:42:07.074 I ns/openshift-marketplace pod/csctestlabel-cb548bf6f-xp9qg Container image "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:9fd355d207a75d6944223c315e3dee803fbacf071912359909f0d443a5e6efec" already present on machine
Dec 02 02:42:07.346 I ns/openshift-marketplace pod/csctestlabel-674c7856bb-z42qw Container image "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:9fd355d207a75d6944223c315e3dee803fbacf071912359909f0d443a5e6efec" already present on machine
Dec 02 02:42:07.460 I ns/openshift-marketplace pod/csctestlabel-cb548bf6f-xp9qg Created container csctestlabel
Dec 02 02:42:07.501 I ns/openshift-marketplace pod/csctestlabel-cb548bf6f-xp9qg Started container csctestlabel
Dec 02 02:42:07.525 I ns/openshift-marketplace pod/csctestlabel-674c7856bb-z42qw Created container csctestlabel
Dec 02 02:42:07.566 I ns/openshift-marketplace pod/csctestlabel-674c7856bb-z42qw Started container csctestlabel
Dec 02 02:42:12.909 I ns/openshift-marketplace deployment/csctestlabel Scaled down replica set csctestlabel-cb548bf6f to 0
Dec 02 02:42:12.912 W ns/openshift-marketplace pod/csctestlabel-cb548bf6f-xp9qg node/ip-10-0-159-84.us-east-2.compute.internal graceful deletion within 30s
Dec 02 02:42:12.995 I ns/openshift-marketplace replicaset/csctestlabel-cb548bf6f Deleted pod: csctestlabel-cb548bf6f-xp9qg
Dec 02 02:42:13.201 I ns/openshift-marketplace pod/csctestlabel-cb548bf6f-xp9qg Stopping container csctestlabel
Dec 02 02:42:16.759 W ns/openshift-marketplace pod/opsrctestlabel-7784d95d48-94m2j node/ip-10-0-159-84.us-east-2.compute.internal graceful deletion within 30s
Dec 02 02:42:16.771 I ns/openshift-marketplace pod/opsrctestlabel-7784d95d48-94m2j Stopping container opsrctestlabel
Dec 02 02:42:17.005 W ns/openshift-marketplace pod/csctestlabel-cb548bf6f-xp9qg node/ip-10-0-159-84.us-east-2.compute.internal invariant violation (bug): pod should not transition Running->Pending even when terminated
Dec 02 02:42:17.913 E ns/openshift-marketplace pod/opsrctestlabel-7784d95d48-94m2j node/ip-10-0-159-84.us-east-2.compute.internal container=opsrctestlabel container exited with code 2 (Error): 
Dec 02 02:42:18.160 W ns/openshift-marketplace pod/csctestlabel-674c7856bb-z42qw node/ip-10-0-131-240.us-east-2.compute.internal graceful deletion within 30s
Dec 02 02:42:18.185 I ns/openshift-marketplace pod/csctestlabel-674c7856bb-z42qw Stopping container csctestlabel

Writing JUnit report to junit_e2e_20191202-024218.xml

1 pass, 0 skip (55.6s)
```

jianzhangbjz commented:
@kevinrizza Could you help take a look at the above issue? Thanks very much!

jianzhangbjz commented:
/retest

jianzhangbjz commented:
/retest

bparees commented:
/assign @shawn-hurley

shawn-hurley commented:
/approve

kevinrizza commented on test/extended/testdata/olm/image-catalogsource.yaml:
```diff
@@ -0,0 +1,10 @@
+  namespace: openshift-marketplace
+spec:
+  sourceType: grpc
+  image: quay.io/aosqe42/operator-registry:auto
```
What publishes this image? Is it easily reproducible? Should it come from a build system?

jianzhangbjz replied:
It was build by using the `operator-registry`. As follows:
```console
1, git clone https://github.com/operator-framework/operator-registry

2, mac:operator-registry jianzhang$ cd manifests/
mac:manifests jianzhang$ ls
etcd                   prometheus             strimzi-kafka-operator
mac:manifests jianzhang$ rm -rf prometheus strimzi-kafka-operator
mac:manifests jianzhang$ 
3, mac:manifests jianzhang$ tree
.
└── etcd
    ├── 0.6.1
    │   ├── etcdcluster.crd.yaml
    │   └── etcdoperator.clusterserviceversion.yaml
    ├── 0.9.2
    │   ├── etcdbackup.crd.yaml
    │   ├── etcdcluster.crd.yaml
    │   ├── etcdoperator.v0.9.0.yaml
    │   ├── etcdoperator.v0.9.2.clusterserviceversion.yaml
    │   └── etcdrestore.crd.yaml
    └── etcd.package.yaml

3 directories, 8 files

4, mac:operator-registry jianzhang$ docker build -f upstream-example.Dockerfile  -t quay.io/aosqe42/operator-registry:auto2 .
Sending build context to Docker daemon  81.88MB
```

jianzhangbjz replied:
```console
mac:operator-registry jianzhang$ docker push quay.io/aosqe42/operator-registry:auto2 
```

jianzhangbjz replied:
Here is the related manifest: https://github.com/jianzhangbjz/operator-registry/tree/pr-23186/manifests/etcd-test

kevinrizza replied:
So I think relying on a locally built image for this test is somewhat problematic. Can we try to use a reproducible image based on a build environment? What happens when something about this image needs to/does change and we don't have automation in place to regenerate this image? Who owns this namespace and how can we update the images in it? Can we at the very least have some common credentials and a common build environment so that we can migrate this over time? At the very least some documentation so that someone can decipher how to do this two years from now if none of us are around anymore?

kevinrizza replied:
Alternatively, could we just use a config map for this test for now? We have been thinking about how it would be possible to generate these images at test time, but right now the build environment doesn't have an obvious way to generate a container image that can be pushed to a registry while a test is actually running.

njhale commented on test/extended/operators/olm.go:
```diff
@@ -87,4 +91,66 @@ var _ = g.Describe("[Feature:Platform] OLM should", func() {
+		for k := range getCmds {
+			keys = append(keys, k)
+		}
+		sort.Ints(keys)
```
I can't see a reason to use a map here. It adds a lot of extra code too. A simple alternative would be to use a slice of slices:

```go
getCmds := [][]string{
    {"subscription", "test-operator"},
    {"clusterserviceversion", "etcdoperator.v0.9.2"},
    {"pods", "etcd-operator"},
}

for _, cmd := range getCmds {
    err := wait.Poll(3*time.Second, 120*time.Second, func() (bool, error) {
        output, err := oc.AsAdmin().WithoutNamespace().Run("get").Args(cmd[0], "-n", "openshift-operators").Output()
        if err != nil {
	    e2e.Logf("Fail to get %v, error:%v", cmd[0], err)
	    return false, nil
        }
        if strings.Contains(output, cmd[1]) {
            return true, nil
        }
        return false, nil
    })
    o.Expect(err).NotTo(o.HaveOccurred())
}
```

If you want something more foolproof, you can use a slice of structs (what I **strongly** recommend):

```go
getCmds := []struct{
    kind
    expect
}{
    {"subscription", "test-operator"},
    {"clusterserviceversion", "etcdoperator.v0.9.2"},
    {"pods", "etcd-operator"},
}

for _, cmd := range getCmds {
    err := wait.Poll(3*time.Second, 120*time.Second, func() (bool, error) {
        output, err := oc.AsAdmin().WithoutNamespace().Run("get").Args(cmd.kind, "-n", "openshift-operators").Output()
        if err != nil {
	    e2e.Logf("Fail to get %v, error:%v", cmd.kind, err)
	    return false, nil
        }
        if strings.Contains(output, cmd.expect) {
            return true, nil
        }
        return false, nil
    })
    o.Expect(err).NotTo(o.HaveOccurred())
}
```

jianzhangbjz replied:
Thanks very much! Updated it.

jianzhangbjz commented on test/extended/operators/olm.go:
```diff
@@ -87,4 +91,66 @@ var _ = g.Describe("[Feature:Platform] OLM should", func() {
+		_, err := oc.AsAdmin().WithoutNamespace().Run("create").Args("-f", catalogsourceYAML).Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		err = wait.Poll(3*time.Second, 120*time.Second, func() (bool, error) {
+			output, err := oc.AsAdmin().WithoutNamespace().Run("get").Args("packagemanifest").Output()
```
Yes, you're right. Updated it.

jianzhangbjz replied:
Actually, there two `packagemanifests`, both of them called `etcd`. One is from the `Community Operators` catalog, and another one is from this `Test Operators` catalog. Anyway, I think I'd better to make other names for this test `etcd`.

jianzhangbjz commented on test/extended/testdata/olm/image-sub.yaml:
```diff
@@ -0,0 +1,11 @@
+ source: test-operator
+ sourceNamespace: openshift-marketplace
+ name: etcd
+ startingCSV: etcdoperator.v0.9.2
```
Thanks! I will update it with the latest csv.

jianzhangbjz commented on test/extended/testdata/olm/image-catalogsource.yaml:
```diff
@@ -0,0 +1,10 @@
+  namespace: openshift-marketplace
+spec:
+  sourceType: grpc
+  image: quay.io/aosqe42/operator-registry:auto
```
@njhale How to do that?

jianzhangbjz replied:
@njhale Could you give some suggestions? Thanks!

njhale commented:
This looks a lot cleaner. Thank you for taking the time to refine this!

/lgtm

jianzhangbjz commented:
/retest

jianzhangbjz commented:
/test e2e-gcp-upgrade
/test e2e-aws-fips

jianzhangbjz commented:
/retest

jianzhangbjz commented:
@njhale Your suggestions help a lot! Thanks! @shawn-hurley Could you help add the approved label? Thanks!

shawn-hurley commented:
/retest
/approve

shawn-hurley commented:
/retest

shawn-hurley commented:
Seems these are failing due to: 
https://bugzilla.redhat.com/show_bug.cgi?id=1781708

/retest

jianzhangbjz commented:
/retest

jianzhangbjz commented:
/test e2e-gcp-upgrade

jianzhangbjz commented:
Rebase it. @njhale Could you help add the `lgtm` label again? Thanks!

jianzhangbjz commented:
/retest

jianzhangbjz commented:
/retest

jianzhangbjz commented:
/retest

jianzhangbjz commented:
/retest

jianzhangbjz commented:
/retest

emmajiafan commented:
/lgtm

jianzhangbjz commented:
/test e2e-gcp-upgrade

jianzhangbjz commented:
/retest

jianzhangbjz commented:
Automate it in: openshift/openshift-tests-private#75, close this PR.
